app/upgrades/v4/upgrade_test.go

```diff
@@ -0,0 +1,105 @@
+package v4_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	"github.com/stretchr/testify/suite"
+	abci "github.com/tendermint/tendermint/abci/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+
+	"github.com/dymensionxyz/dymension/v3/app"
+	"github.com/dymensionxyz/dymension/v3/app/apptesting"
+)
+
+// UpgradeTestSuite defines the structure for the upgrade test suite
+type UpgradeTestSuite struct {
+	suite.Suite
+	Ctx sdk.Context
+	App *app.App
+}
+
+// SetupTest initializes the necessary items for each test
+func (s *UpgradeTestSuite) SetupTest(t *testing.T) {
+	s.App = apptesting.Setup(t, false)
+	s.Ctx = s.App.BaseApp.NewContext(false, tmproto.Header{Height: 1, ChainID: "dymension_100-1", Time: time.Now().UTC()})
+}
+
+// TestUpgradeTestSuite runs the suite of tests for the upgrade handler
+func TestUpgradeTestSuite(t *testing.T) {
+	suite.Run(t, new(UpgradeTestSuite))
+}
+
+var (
+	dummyUpgradeHeight                    int64 = 5
+	expectDelayedackDeletePacketBatchSize int32 = 1000
+	expectDelayedackEpochIdentifier             = "hour"
+	expectDelayedackBridgingFee                 = sdk.NewDecWithPrec(1, 3)
+)
+
+// TestUpgrade is a method of UpgradeTestSuite to test the upgrade process.
+func (s *UpgradeTestSuite) TestUpgrade() {
+	testCases := []struct {
+		msg         string
+		upgrade     func()
+		postUpgrade func() error
+		expPass     bool
+	}{
+		{
+			"Test that upgrade does not panic and sets correct parameters",
```

Comment:
dont need 'test that'

In general you dont need to repeat contextually obvious info all the time